nucliadb_core/src/metrics/metrics/tokio_tasks.rs

```diff
@@ -0,0 +1,370 @@
+     are first polled",
+);
+
+const TOTAL_IDLED_COUNT: (&str, &str) = (
+    "nucliadb_node_total_idled_count",
+    "The total number of times that tasks idled, waiting to be awoken. [...]",
+);
+
+const TOTAL_IDLE_DURATION: (&str, &str) = (
+    "nucliadb_node_total_idle_duration",
+    "The total duration that tasks idled. [...]",
+);
+
+const TOTAL_SCHEDULED_COUNT: (&str, &str) = (
+    "nucliadb_node_total_scheduled_count",
+    "The total number of times that tasks were awoken (and then, presumably, scheduled for \
+     execution)",
+);
+
+const TOTAL_SCHEDULED_DURATION: (&str, &str) = (
+    "nucliadb_node_total_scheduled_duration",
+    "The total duration that tasks spent waiting to be polled after awakening",
+);
+
+const TOTAL_POLL_COUNT: (&str, &str) = (
+    "nucliadb_node_total_poll_count",
+    "The total number of times that tasks where polled",
+);
+
+const TOTAL_POLL_DURATION: (&str, &str) = (
+    "nucliadb_node_total_poll_duration",
+    "The total duration elapsed during polls",
+);
+
+const TOTAL_FAST_POLL_COUNT: (&str, &str) = (
+    "nucliadb_node_total_fast_poll_count",
+    "The total number of times that polling tasks completed swiftly. [...]",
+);
+
+const TOTAL_FAST_POLL_DURATION: (&str, &str) = (
+    "nucliadb_node_total_fast_poll_duration",
+    "The total duration of fast polls. [...]",
+);
+
+const TOTAL_SLOW_POLL_COUNT: (&str, &str) = (
+    "nucliadb_node_total_slow_poll_count",
+    "The total number of times that polling tasks completed slowly. [...]",
+);
+
+const TOTAL_SLOW_POLL_DURATION: (&str, &str) = (
+    "nucliadb_node_total_slow_poll_duration",
+    "The total duration of slow polls. [...]",
+);
+
+const TOTAL_SHORT_DELAY_COUNT: (&str, &str) = (
+    "nucliadb_node_total_short_delay_count",
+    "The total count of tasks with short scheduling delays. [...]",
+);
+
+const TOTAL_LONG_DELAY_COUNT: (&str, &str) = (
+    "nucliadb_node_total_long_delay_count",
+    "The total count of tasks with long scheduling delays. [...]",
+);
+
+const TOTAL_SHORT_DELAY_DURATION: (&str, &str) = (
+    "nucliadb_node_total_short_delay_duration",
+    "The total duration of tasks with short scheduling delays. [...]",
+);
+
+const TOTAL_LONG_DELAY_DURATION: (&str, &str) = (
+    "nucliadb_node_total_long_delay_duration",
+    "The total duration of tasks with long scheduling delays. [...]",
+);
+
+#[derive(Clone, Debug, Hash, PartialEq, Eq, EncodeLabelSet)]
+pub struct TaskLabels {
+    pub request: String, // gRPC method (/NewShard, /SetResource...)
+}
+
+// Tasks in tokio should never block longer than 10-100µs
+
+// TODO we are trying bucket values for everything. After an evaluation on
+// production, we should reconsider changing them and customize for every
+// Histogram metric
+const BUCKETS: [f64; 15] = [
+    0.000010, 0.000025, 0.000050, 0.000100, 0.000250, 0.000500, 0.001, 0.002, 0.005, 0.010, 0.100,
+    0.250, 0.500, 1.0, 5.0,
+];
+
+pub struct TokioTaskMetrics {
```

Comment:
This looks like something that would benefit from a macro..
Maybe in the future we can look into it.

Reply:
If you refer to the buckets, there's some functions from prometheus_client that can create linear or exponential bucket values.

For repeated code in metrics, a simple macro could be nice!